### DISCUSSION TITLE
Adaptive wrap prefix?

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#23936] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

The feature request is like https://github.com/zed-industries/zed/issues/98 , but for text / markdown files.

See the screenshots for illustration. The name "Adaptive wrap prefix" comes from Emacs.

### If applicable, add mockups / screenshots to help present your vision of the feature

Zed
<img width="606" alt="Screenshot 2024-02-29 at 23 07 53" src="https://github.com/zed-industries/zed/assets/5475262/3746028f-6f32-4139-b33e-a3c1b3680955">

Emacs 
<img width="952" alt="Screenshot 2024-02-29 at 23 05 27" src="https://github.com/zed-industries/zed/assets/5475262/7227dd2e-b5eb-408b-9bcb-ab53b8147249">

**Comment by maxbrunsfeld** [comment]
@failable I converted this one to a discussion, since I think it needs some design decisions for how exactly we want it to work in Zed.
